### DISCUSSION TITLE
[Help] Implementing Multimon in Wayland

### DISCUSSION THREAD

**Discussion by cmidkiff87** [FreeRDP/FreeRDP#7528] [General]
I'm trying to implement Multi-monitor support in Wayland (issue #7181); however, I've not done anything with Linux GUI (much less Wayland itself) before. 

I have some ideas
1. Create multiple top level surfaces, one for each display, and full-screen each individually
2. Create one top level surface, with one [sub-surface](https://wayland.app/protocols/wayland#wl_subsurface) for each display, and full-screen each subsurface.
3. Create one top level surface, and open ["popups"](https://wayland.app/protocols/xdg-shell#xdg_popup) for each display.

I am unsure of the viability of options 2 and 3.  Does wayland support fullscreening subsurfaces? And would popups work as intended if the top level surface itself isn't in the fullscreen state?

Option 1 [seems to be the most straightforward](https://doc.qt.io/qt-6/qtwaylandcompositor-spanning-screens-example.html) Wayland side, but it looks to require the most code restructuring.

Thoughts?

**Comment by akallabeth** [comment]
@cmidkiff87 I´m also not (too) familiar with the specific internals to support multimon. I think you´re best off using one surface per monitor (`xfreerdp` does it in a similar fashion).
Other than that I can only point you to the wayland protocol documentation https://wayland-book.com/ and hope you can find your answers there.
For `FreeRDP` specific issues I´m usually in our chat (see https://freerdp.com for details) and if not try to answer within a day or two.
